packages/edit-site/src/components/global-styles/screen-block.js

```diff
@@ -150,17 +151,17 @@ function ScreenBlock( { name, variation } ) {
 	const styleWithLayout = useMemo( () => {
 		return {
 			...style,
-			layout: settings.layout,
+			layout: userSettings.layout,
 		};
-	}, [ style, settings.layout ] );
+	}, [ style, userSettings.layout ] );
 	const onChangeDimensions = ( newStyle ) => {
 		const updatedStyle = { ...newStyle };
 		delete updatedStyle.layout;
 		setStyle( updatedStyle );
 
 		if ( newStyle.layout !== settings.layout ) {
 			setSettings( {
-				...rawSettings,
+				...userSettings,
 				layout: newStyle.layout,
 			} );
```

Comment:
In practice, when is this code ever reached? Do we expose layout controls anywhere at the block level in global styles? I don't think we do, but please let me know if I've missed something! The code change here looks good, but I couldn't think of an area where it's currently exposed, so I haven't tested this code path.

Reply:
Is there any chance this is in preparation for something like section-specific theme.json?

Reply:
Ah, that could make sense 👍

Reply:
This code _is_ being reached, though it shouldn't 😅 because even if the contents of `newStyle.layout` and `settings.layout` were the same, they're different objects so that condition is always true.

We could do a deep equality comparison but in that case it would be better to compare `newStyle.layout` with `userSettings.layout`, because `settings.layout` for the blocks is always `
{contentSize: false, wideSize: false}` (I assume because the blocks don't currently support those settings)

We don't currently expose layout controls at block level, though that could conceivably change (content/wide size could perhaps be set at global level for certain blocks?). Even so it might be better to scrap all the layout-specific logic in this screen for now and add it in later if/when we need it.